abstract/encoding.go

```diff
@@ -170,13 +193,31 @@ func (de *decoder) value(v reflect.Value, depth int) error {
 			}
 		}
 
-	case reflect.Slice, reflect.Array:
+	case reflect.Slice:
+		if v.IsNil() {
+			panic("slices must be initialized to correct length before decoding")
+		}
+		fallthrough
+	case reflect.Array:
 		l := v.Len()
 		for i := 0; i < l; i++ {
 			if err = de.value(v.Index(i), depth+1); err != nil {
 				return err
 			}
 		}
+
+	case reflect.Int:
```

Comment:
That is what we discussed also yesterday. Going with the Int32 approach, that would raise error only on the "sender" part, and not on the "receiver" part. If a server is using a big number and wants to encode it as an Int (so implicitly as an Int32), it will fail **before** it has a chance to send it. Better send nothing than send something that will potentially fail, I'd say.